src/ssids/cpu/BuddyAllocator.hxx

```diff
@@ -101,8 +101,6 @@ public:
 #endif /* MEM_STATS */
    }
    ~Page() noexcept(false) {
-      if(next_ && head_[nlevel-1] != 0)
-         throw std::runtime_error("outstanding allocations on cleanup\n");
```

Comment:
I do not understand why this can just be removed and how it relates to the other changes.

Reply:
Is it safe to just remove? To be honest, I do not know. I just know that it did not crash when I removed this check, so I can only hope it did not corrupt memory somewhere.

What I do know is that if I do **not** remove this check, then the exception I throw with the `if (t >= BLOCK_SIZE || m >= BLOCK_SIZE) throw SingularError(p);` in `block_ldlt.hxx` causes `~Page` to be run in this `(next_ && head_[nlevel-1] != 0)` state, so it throws that `std::runtime_error`, and `std::terminate` gets called as a result. So **something** has to be done about that.

If removing the check is not safe, then we need another way to handle this.

Reply:
It **might** also be that existing exceptions thrown in `block_ldlt.hxx` by this check:
```c++
      if(fabs(bestv) < small) {
         if(!action) throw SingularError(p);
```
can also trigger this error case, but I am not sure about that either. It might also be that it only gets triggered because I am throwing an exception also in the case of a non-zero `action` (i.e., that that is the context in which the exception is unexpected). I do not understand the allocator usage well enough yet to know the answer.